src/snowflake/snowpark/mock/analyzer.py

```diff
@@ -268,7 +273,7 @@ def analyze(
             sql = str(expr.value)
             if parse_local_name:
                 sql = sql.upper()
-            return sql
+            return f"'{sql}'" if quote_literal else sql
```

Comment:
Why do we need this change?

Reply:
I added some context in docstring and comment

Reply:
Do we need to worry if this is already quoted?

Reply:
^, I guess it happens when a string literal `'"hello"'`.

Reply:
updated to use `calculate_expression` to literal computation and single quote literal all the time in analyze.